compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1407,7 +1407,7 @@ public BType transform(BLangTableTypeNode tableTypeNode, AnalyzerData data) {
     @Override
     public BType transform(BLangFiniteTypeNode finiteTypeNode, AnalyzerData data) {
         BTypeSymbol finiteTypeSymbol = Symbols.createTypeSymbol(SymTag.FINITE_TYPE,
-                Flags.asMask(EnumSet.noneOf(Flag.class)), Names.EMPTY,
+                Flags.asMask(EnumSet.of(Flag.PUBLIC)), Names.EMPTY,
```

Comment:
This is because 
```bal
resource function put books/["books" a]() returns string {
    return a;
}
```
Gives `attempt to expose non-public symbol ''` to `a` after making the resource function public

```bal
public "a" a = "a";
```
this also gives the same error and fixed with above fix @MaryamZi @pcnfernando @gimantha Please verify this

Reply:
Verified offline